analytics/src/main/java/com/segment/analytics/ConnectionFactory.java

```diff
@@ -54,9 +54,15 @@ public HttpURLConnection projectSettings(String writeKey) throws IOException {
      * Return a {@link HttpURLConnection} that writes batched payloads to {@code
      * https://api.segment.io/v1/import}.
      */
-    public HttpURLConnection upload(String apiHost, String writeKey) throws IOException {
+    public HttpURLConnection upload(String apiHost, String key, boolean useSigning) throws IOException {
         HttpURLConnection connection = openConnection(String.format("https://%s/import", apiHost));
-        connection.setRequestProperty("Authorization", authorizationHeader(writeKey));
+        if (useSigning) {
+            connection.setRequestProperty("Authorization", "SEGMENT-HMAC-SHA1");
+            connection.setRequestProperty("Signature", key);
```

Comment:
also the header key should be `X-Segment-Signature`

Reply:
discussed in slack